### PR TITLE
29880 - No completions for computed string properties

### LINKED ISSUE #29880: No completions for computed string properties
```ts
const otherThing = "otherThing";

interface Foo {
    ["thing"]: number;
    [otherThing]: string;
}

declare var x: Foo;
x./**/
```

**Expected**: completions for `thing` and `[otherThing]`.
**Actual**: completions for only `[otherThing]`

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #29880

### CODE REVIEW

File: tests/cases/fourslash/completionForComputedStringProperties.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+//// const p2 = "p2";
+//// interface A {
+////     ["p1"]: string;
+////     [p2]: string;
+//// }
+//// declare const a: A;
+//// a./*1*/ ;
+//// a["/*2*/ ;
+
+verify.completions({ marker: "1", exact: "p1" });
```
Comment (DanielRosenwasser) [inline tests/cases/fourslash/completionForComputedStringProperties.ts]: Shouldn't `p2` show up here?

File: src/services/completions.ts [hunk 2]
```diff
@@ -965,6 +965,9 @@ namespace ts.Completions {
                     symbolToOriginInfoMap[getSymbolId(firstAccessibleSymbol)] =
                         !moduleSymbol || !isExternalModuleSymbol(moduleSymbol) ? { kind: SymbolOriginInfoKind.SymbolMemberNoExport } : { kind: SymbolOriginInfoKind.SymbolMemberExport, moduleSymbol, isDefaultExport: false };
                 }
+                else {
+                    symbols.push(symbol);
```
Comment (DanielRosenwasser) [inline src/services/completions.ts]: Seems like you could just have a `symbols.push(symbol)` at the end and use an early return in the inner-most `if`

Comment (a-tarasyuk) [reply src/services/completions.ts]: @DanielRosenwasser done.

File: tests/cases/fourslash/completionForComputedStringProperties.ts (hunk 1 above)
Comment (a-tarasyuk) [reply tests/cases/fourslash/completionForComputedStringProperties.ts]: @DanielRosenwasser Maybe you can clarify and point me out how to trigger in `verify.completions` list for `.` and `[]` which will contain all available properties. For instance server returns both properties, however, behave differently



when choosing dynamic declared property dot will be replaced by `[]`. Maybe for that reason, the first version doesn't contain dynamic declared property.

Comment (a-tarasyuk) [reply tests/cases/fourslash/completionForComputedStringProperties.ts]: @DanielRosenwasser what do you think?

Comment (sheetalkamat) [reply tests/cases/fourslash/completionForComputedStringProperties.ts]: I think you need to pass user preferences with `includeInsertTextCompletions` set to `true`. Eg check `tests/cases/fourslash/completionsThisType.ts `

Comment (a-tarasyuk) [reply tests/cases/fourslash/completionForComputedStringProperties.ts]: @sheetalkamat thanks ;)

File: src/services/completions.ts [hunk 3]
```diff
@@ -964,11 +964,11 @@ namespace ts.Completions {
                     const moduleSymbol = firstAccessibleSymbol.parent;
                     symbolToOriginInfoMap[getSymbolId(firstAccessibleSymbol)] =
                         !moduleSymbol || !isExternalModuleSymbol(moduleSymbol) ? { kind: SymbolOriginInfoKind.SymbolMemberNoExport } : { kind: SymbolOriginInfoKind.SymbolMemberExport, moduleSymbol, isDefaultExport: false };
+                    return;
                 }
```
Comment (sheetalkamat) [inline src/services/completions.ts]: I think in the else part you want to check `includeInsertTextCompletions` to be true and only then add symbol

Comment (a-tarasyuk) [reply src/services/completions.ts]: @samelhusseini do you mean `preferences.includeInsertTextCompletions`?

```typescript
if (preferences.includeCompletionsWithInsertText) {
   symbols.push(symbol);
}
```

Comment (sheetalkamat) [reply src/services/completions.ts]: Yes

Comment (a-tarasyuk) [reply src/services/completions.ts]: @sheetalkamat oke, thanks 👍

Comment (a-tarasyuk) [reply src/services/completions.ts]: @sheetalkamat Could you review this PR again? Thanks in advance.

**Comment by DanielRosenwasser** [comment]
Thank you!

**Pull request merged**